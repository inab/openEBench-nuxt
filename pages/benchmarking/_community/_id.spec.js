import { mount } from '@vue/test-utils';
import Challenge from './_id.vue';
import MockCommunity from '~/test/unit/mockData/Community';
import MockEvent from '~/test/unit/mockData/Event';
import MockEvents from '~/test/unit/mockData/Events';
import MockChallenge from '~/test/unit/mockData/Challenge';
import MockChallengeDatasetsBarplot from '~/test/unit/mockData/ChallengeDatasetsBarplot';
import MockChallengeDatasetsScatter from '~/test/unit/mockData/ChallengeDatasetsScatter';

const factory = (mockStore) => {
	return mount(Challenge, {
		...createComponentMocks({ store: mockStore }),
		mocks: {
			$route: { params: { community: 'TESTCOMMUNITY', id: 'TESTID' } },
			$config: { OEB_LEGACY_ANGULAR_URI: 'https://jest-openebench.bsc.es/' },
		},
	});
};

describe('Community Participant', () => {
	afterEach(() => {
		jest.clearAllMocks();
	});

	const mockStore = {
		community: {
			getters: {
				events: () => {
					return MockEvents;
				},
				currentEvent: () => {
					return MockEvent;
				},
				datasets: () => {
					return [];
				},
				tools: () => {
					return [];
				},
				community: () => {
					return MockCommunity;
				},
				getEventById: () => {
					return () => MockEvent;
				},
			},
			actions: {
				setCurrentEvent: jest.fn(),
				getTools: jest.fn(),
				getDatasets: jest.fn(),
				getBenchmarkingEvents: jest.fn(),
				getCommunity: jest.fn(),
			},
			mutations: {
				setCommunity: jest.fn(),
				setEvents: jest.fn(),
				setCurrentEvent: jest.fn(),
				setDatasets: jest.fn(),
				setTools: jest.fn(),
				setLoading: jest.fn(),
			},
			state: () => {
				return {
					loading: {
						events: false,
						tools: false,
						datasets: false,
						community: false,
					},
					community: MockCommunity,
				};
			},
		},
		challenge: {
			getters: {
				datasetsList: () => {
					return [];
				},
				challenge: () => {
					return {};
				},
			},
			actions: {
				getChallenge: jest.fn(),
				getDatasets: jest.fn(),
			},
			mutations: {
				setDatasets: jest.fn(),
				setChallenge: jest.fn(),
				setLoading: jest.fn(),
			},
			state: () => {
				return {
					loading: {
						challenge: false,
						datasets: false,
					},
					challenge: MockChallenge,
				};
			},
		},
	};

	it('is instantiated', async () => {
		const wrapper = factory(mockStore);
		expect(wrapper).toBeTruthy();
		await wrapper.vm.$nextTick();
	});

	it('renders the bar-plot component', () => {
		mockStore.challenge.getters.datasetsList = () => {
			return MockChallengeDatasetsBarplot;
		};
		const wrapper = factory(mockStore);
		expect(wrapper).toBeTruthy();
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('renders the scatter-plot component', () => {
		mockStore.challenge.getters.datasetsList = () => {
			return MockChallengeDatasetsScatter;
		};
		const wrapper = factory(mockStore);
		expect(wrapper).toBeTruthy();
		expect(wrapper.html()).toMatchSnapshot();
	});
});
