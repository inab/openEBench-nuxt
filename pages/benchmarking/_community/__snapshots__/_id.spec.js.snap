// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community Participant renders the bar-plot component 1`] = `
"<div class=\\"container container--fluid\\">
  <div>
    <h1 class=\\"text-h4\\">

    </h1>
    <h2 class=\\"text-subtitle-1 mb-5\\">

    </h2>
    <p class=\\"text--secondary\\">
      In this 2D plot two metrics from the challenge
      are represented in the X and Y axis, showing the
      results from the participating tools in this challenge. The gray line
      represents the pareto frontier, which runs over the participants showing
      the best efficiency and the arrow in the plot represents the optimal
      corner.
    </p>
    <div role=\\"alert\\" class=\\"v-alert mt-8 v-sheet theme--dark v-alert--border v-alert--dense v-alert--text v-alert--border-left info--text\\">
      <div class=\\"v-alert__wrapper\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-alert__icon mdi mdi-information theme--dark info--text\\"></i>
        <div class=\\"v-alert__content\\">
          The menu button above the diagram can be used to switch between the
          different classification methods / visualization modes (Square
          Quartiles; Diagonal Quartiles, and k-means Clustering).
        </div>
        <div class=\\"v-alert__border v-alert__border--left\\"></div>
      </div>
    </div>
  </div>
  <div>
    <!---->
    <div class=\\"mt-2 v-item-group theme--light v-slide-group v-chip-group v-chip-group--column\\">
      <div class=\\"v-slide-group__prev v-slide-group__prev--disabled\\">
        <transition-stub name=\\"fade-transition\\"></transition-stub>
      </div>
      <div class=\\"v-slide-group__wrapper\\">
        <div class=\\"v-slide-group__content\\"><span tabindex=\\"0\\" class=\\"v-chip primary--text v-chip--active v-chip--clickable v-chip--no-color theme--light v-size--default\\"><span class=\\"v-chip__content\\">
				Weighted Probabilistic Concordance Index (WPC-i)
			</span></span></div>
      </div>
      <div class=\\"v-slide-group__next v-slide-group__next--disabled\\">
        <transition-stub name=\\"fade-transition\\"></transition-stub>
      </div>
    </div>
    <div class=\\"v-window v-item-group theme--light v-tabs-items\\">
      <div class=\\"v-window__container\\">
        <transition-stub name=\\"\\">
          <div class=\\"v-window-item v-window-item--active\\">
            <div data-id=\\"OEBD003000002S\\" data-api-url=\\"undefined/graphql\\" class=\\"benchmarkingChart_bars mt-5\\"></div>
          </div>
        </transition-stub>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Community Participant renders the scatter-plot component 1`] = `
"<div class=\\"container container--fluid\\">
  <div>
    <h1 class=\\"text-h4\\">

    </h1>
    <h2 class=\\"text-subtitle-1 mb-5\\">

    </h2>
    <p class=\\"text--secondary\\">
      In this 2D plot two metrics from the challenge
      are represented in the X and Y axis, showing the
      results from the participating tools in this challenge. The gray line
      represents the pareto frontier, which runs over the participants showing
      the best efficiency and the arrow in the plot represents the optimal
      corner.
    </p>
    <div role=\\"alert\\" class=\\"v-alert mt-8 v-sheet theme--dark v-alert--border v-alert--dense v-alert--text v-alert--border-left info--text\\">
      <div class=\\"v-alert__wrapper\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-alert__icon mdi mdi-information theme--dark info--text\\"></i>
        <div class=\\"v-alert__content\\">
          The menu button above the diagram can be used to switch between the
          different classification methods / visualization modes (Square
          Quartiles; Diagonal Quartiles, and k-means Clustering).
        </div>
        <div class=\\"v-alert__border v-alert__border--left\\"></div>
      </div>
    </div>
  </div>
  <div>
    <h2 class=\\"text-h6 mt-10\\">Choose dataset:</h2>
    <div class=\\"mt-2 v-item-group theme--light v-slide-group v-chip-group v-chip-group--column\\">
      <div class=\\"v-slide-group__prev v-slide-group__prev--disabled\\">
        <transition-stub name=\\"fade-transition\\"></transition-stub>
      </div>
      <div class=\\"v-slide-group__wrapper\\">
        <div class=\\"v-slide-group__content\\"><span tabindex=\\"0\\" class=\\"v-chip primary--text v-chip--active v-chip--clickable v-chip--no-color theme--light v-size--default\\"><span class=\\"v-chip__content\\">
				Fraction returned + Average CAD-score
			</span></span><span tabindex=\\"0\\" class=\\"v-chip v-chip--clickable v-chip--no-color theme--light v-size--default\\"><span class=\\"v-chip__content\\">
				Fraction returned + Average lDDT
			</span></span><span tabindex=\\"0\\" class=\\"v-chip v-chip--clickable v-chip--no-color theme--light v-size--default\\"><span class=\\"v-chip__content\\">
				Fraction returned + Average QS-score
			</span></span><span tabindex=\\"0\\" class=\\"v-chip v-chip--clickable v-chip--no-color theme--light v-size--default\\"><span class=\\"v-chip__content\\">
				Fraction returned + Average MM-align TM-score
			</span></span><span tabindex=\\"0\\" class=\\"v-chip v-chip--clickable v-chip--no-color theme--light v-size--default\\"><span class=\\"v-chip__content\\">
				Fraction returned + Average lDDT-BS
			</span></span><span tabindex=\\"0\\" class=\\"v-chip v-chip--clickable v-chip--no-color theme--light v-size--default\\"><span class=\\"v-chip__content\\">
				Fraction returned + Average Model confidence
			</span></span></div>
      </div>
      <div class=\\"v-slide-group__next v-slide-group__next--disabled\\">
        <transition-stub name=\\"fade-transition\\"></transition-stub>
      </div>
    </div>
    <div class=\\"v-window v-item-group theme--light v-tabs-items\\">
      <div class=\\"v-window__container\\">
        <transition-stub name=\\"\\">
          <div class=\\"v-window-item v-window-item--active\\">
            <div data-id=\\"OEBD0080000U6A\\" data-api-url=\\"undefined/graphql\\" totable=\\"true\\" data-mode=\\"openebench\\" class=\\"benchmarkingChart_scatter mt-5\\"></div>
          </div>
        </transition-stub>
        <transition-stub name=\\"\\">
          <!---->
        </transition-stub>
        <transition-stub name=\\"\\">
          <!---->
        </transition-stub>
        <transition-stub name=\\"\\">
          <!---->
        </transition-stub>
        <transition-stub name=\\"\\">
          <!---->
        </transition-stub>
        <transition-stub name=\\"\\">
          <!---->
        </transition-stub>
      </div>
    </div>
  </div>
</div>"
`;
